### PR TITLE
D in visual mode behaves like d

### PR DESCRIPTION
**Pull request by xlaech**
The delete function of the DeleteOperation Class already account for this case, if the Register Mode is set to LineWise.

### CODE REVIEW

**Comment by xlaech** [comment]
Sorry for the earlier Code. Should have created a new branch for the first one as well :)

**Comment by xconverge** [comment]
Nice, mind adding a test to the bottom of modeVisual.test.ts? you can use newTestOnly() to run just the test that you are making, then you can switch it back to newTest() before you commit so that all tests still run

**Comment by xconverge** [comment]
Awesome!

File: src/actions/actions.ts [hunk 1]
```diff
@@ -696,7 +696,9 @@ export class CommandOneNormalCommandInInsertMode extends BaseCommand {
 
   public async exec(position: Position, vimState: VimState): Promise<VimState> {
     vimState.returnToInsertAfterCommand = true;
-    return await new CommandEscInsertMode().exec(position, vimState);
+    return await new CommandEscInsertMode().exec(
```
Comment (johnfn) [inline src/actions/actions.ts]: I don't think that this bit is relevant to the D fix, right? Could you remove it and include it in a separate PR?

Comment (johnfn) [inline src/actions/actions.ts]: Wait nevermind, it doesn't really matter heh.

**Comment by johnfn** [comment]
Excellent, and tests too. Awesome job. Thanks for the PR. :)

**Pull request merged**